src/uniq.js

```diff
@@ -18,4 +20,37 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = (function() {
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  function uniq(list) {
+    /* global Set */
+    var item,
+        set = new Set(),
+        idx = -1,
+        len = list.length,
+        items = [],
+        uniqs = [];
+
+    while (++idx < len) {
```

Comment:
Then in this particular case we will introduce `undefined` in the result. This will become inconsisten with `indexOf` result. Will that be okay?